### PR TITLE
[PWA-26] chore: pwa input focus improvement

### PR DESCRIPTION
**Pull request by anmolsinghbhatia**
### Changes:
This PR includes improvements to PWA input focus. Previously, when opening any dropdown or modal in the PWA, the keyboard would automatically pop up. I have made the necessary adjustments to ensure it now behaves as intended.

### Reference: 
[[PWA-26]](https://app.plane.so/plane/projects/e4b10a8d-1a9a-458b-a546-713d122b1379/issues/afa2c47d-c775-4394-99ee-7e2dd3bd05e8)

## Summary by CodeRabbit

- **New Features**
	- Introduced `isMobile` property across various components to enhance responsiveness based on device type.
	- Updated `tabIndex` management for improved accessibility and navigation on mobile devices.
	- Added dynamic focus behavior for input fields, preventing unwanted focus on mobile.

- **Bug Fixes**
	- Corrected focus behavior by ensuring inputs do not auto-focus on mobile devices.

- **Documentation**
	- Updated documentation to reflect changes in component properties and usage of the `isMobile` state.

- **Chores**
	- Refactored code for better maintainability and clarity, including the introduction of a new `tab-indices` helper for managing tab indices.

### CODE REVIEW

File: web/core/hooks/use-dropdown.ts [hunk 1]
```diff
@@ -12,10 +12,11 @@ type TArguments = {
   query?: string;
   setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
   setQuery?: React.Dispatch<React.SetStateAction<string>>;
+  isMobile?: boolean;
 };
 
 export const useDropdown = (args: TArguments) => {
-  const { dropdownRef, inputRef, isOpen, onClose, onOpen, query, setIsOpen, setQuery } = args;
+  const { dropdownRef, inputRef, isOpen, onClose, onOpen, query, setIsOpen, setQuery, isMobile = false } = args;
```
Comment (gakshita) [inline web/core/hooks/use-dropdown.ts:20]: Why cant we calculate isMobile inside this hook itself

File: web/core/components/views/form.tsx [hunk 2]
```diff
@@ -194,7 +199,7 @@ export const ProjectViewForm: React.FC<Props> = observer((props) => {
                     hasError={Boolean(errors.name)}
                     placeholder="Title"
                     className="w-full text-base"
-                    tabIndex={1}
+                    tabIndex={getTabIndex("name", ETabIndices.PROJECT_VIEW, isMobile)}
```
Comment (gakshita) [inline web/core/components/views/form.tsx:202]: Can you curry the getTabIndex function becuase the isMobile argument is redundant?

File: web/core/components/issues/parent-issues-list-modal.tsx [hunk 3]
```diff
@@ -121,6 +121,7 @@ export const ParentIssuesListModal: React.FC<Props> = ({
                       value={searchTerm}
                       onChange={(e) => setSearchTerm(e.target.value)}
                       displayValue={() => ""}
+                      tabIndex={isMobile ? -1 : 1}
```
Comment (gakshita) [inline web/core/components/issues/parent-issues-list-modal.tsx:124]: Can we move this logic also inside the helper?

File: web/core/components/views/form.tsx (hunk 2 above)
Comment (anmolsinghbhatia) [reply web/core/components/views/form.tsx]: Resolved

File: web/core/components/issues/parent-issues-list-modal.tsx (hunk 3 above)
Comment (anmolsinghbhatia) [reply web/core/components/issues/parent-issues-list-modal.tsx]: Resolved

**Review by gakshita — APPROVED** [review]

**Review by pushya22 — APPROVED** [review]

**Pull request merged**